src/main/kotlin/de/cyface/collector/verticle/CollectorApiVerticle.kt

```diff
@@ -51,90 +49,72 @@ import java.util.Locale
  * @author Armin Schnabel
  * @property authHandlerBuilder Create a handler for authentication requests.
  * @property serverConfiguration The configuration used by this `CollectorApiVerticle`.
- * @property mongoDatabaseConfiguration Configuration of a mongo database to store sensor data and meta data
- * for a measurement.
+ * @property storageServiceBuilder A builder for a DataStorageService, responsible for storing the submitted
+ * information.
  */
 class CollectorApiVerticle(
     private val authHandlerBuilder: AuthHandlerBuilder,
     private val serverConfiguration: ServerConfiguration,
-    private val mongoDatabaseConfiguration: JsonObject
-) : AbstractVerticle() {
+    private val storageServiceBuilder: DataStorageServiceBuilder,
+) : CoroutineVerticle() {
 
     /**
      * Logger used by objects of this class. Configure it using "src/main/resources/logback.xml".
      */
     private val logger: Logger = LoggerFactory.getLogger(CollectorApiVerticle::class.java)
+
+    /**
+     * Creates [de.cyface.collector.model.Measurement] instances from the raw data received by the API.
+     */
     private val measurementFactory = MeasurementFactory()
+
+    /**
+     * Creates [de.cyface.collector.model.Attachment] instances from the raw data received by the API.
+     */
     private val attachmentFactory = AttachmentFactory()
 
     @Throws(Exception::class)
-    override fun start(startPromise: Promise<Void>) {
-        logger.info("Starting collector API!")
-
-        // Start http server
+    override suspend fun start() {
+        try {
+            logger.info("Starting collector API!")
 
-        val mongoClient = MongoClient.createShared(
-            vertx,
-            mongoDatabaseConfiguration,
-            mongoDatabaseConfiguration.getString("data_source_name")
-        )
-        val storageServiceBuilder = serverConfiguration.largeFileStorageType.dataStorageServiceBuilder(
-            vertx,
-            mongoClient
-        )
-        val storageServiceBuilderCall = storageServiceBuilder.create()
-        val serverStartPromise = Promise.promise<Void>()
-        storageServiceBuilderCall.onSuccess { storageService ->
+            // Start http server
+            val storageService = storageServiceBuilder.create().coAwait()
             logger.info("Created storage service.")
             val uploadExpirationTime = serverConfiguration.uploadExpirationTimeInMillis
-            logger.info("Requests to the storage service expire after $uploadExpirationTime milliseconds.")
+            // TODO What does this mean?
+            logger.info("Requests the storage service expire after $uploadExpirationTime milliseconds.")
```

Comment:
The `uploadExpirationTime` determines, when the Cleanup Service removed temporary files. But yeah, this Logging contains typos :-)

Reply:
Ok. Now I think I get it. I hope I updated the documentation and logging so we will be able to understand this in half a year or so.